hera_mc/roach.py

```diff
@@ -0,0 +1,127 @@
+
+
+class RoachTemperature(MCDeclarativeBase):
+    """
+    Definition of roach temperature table.
+
+    time: gps time of the roach data, floored (BigInteger, part of primary_key).
+    roach: roach number (Integer, part of primary_key)
+    ambient_temp: ambient temperature reported by roach in Celcius
+    inlet_temp: inlet temperature reported by roach in Celcius
+    outlet_temp: outlet temperature reported by roach in Celcius
+    fpga_temp: fpga temperature reported by roach in Celcius
+    ppc_temp: ppc temperature reported by roach in Celcius
+    """
+    __tablename__ = 'roach_temperature'
+    time = Column(BigInteger, primary_key=True)
+    roach = Column(Integer, primary_key=True)
+    ambient_temp = Column(Float)
+    inlet_temp = Column(Float)
+    outlet_temp = Column(Float)
+    fpga_temp = Column(Float)
+    ppc_temp = Column(Float)
+
+    @classmethod
+    def create(cls, time, roach, ambient_temp, inlet_temp, outlet_temp, fpga_temp, ppc_temp):
+        """
+        Create a new roach temperature object.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp from the katportal sensor.
+        roach: int
+            roach number
+        ambient_temp: float
+            ambient temperature reported by roach for this time in Celcius
+        inlet_temp: float
+            inlet temperature reported by roach for this time in Celcius
+        outlet_temp: float
+            outlet temperature reported by roach for this time in Celcius
+        fpga_temp: float
+            fpga temperature reported by roach for this time in Celcius
+        ppc_temp: float
+            ppc temperature reported by roach for this time in Celcius
+        """
+        if not isinstance(time, Time):
+            raise ValueError('time must be an astropy Time object')
+        roach_time = floor(time.gps)
+
+        return cls(time=roach_time, roach=roach, ambient_temp=ambient_temp,
+                   inlet_temp=inlet_temp, outlet_temp=outlet_temp,
+                   fpga_temp=fpga_temp, ppc_temp=ppc_temp)
+
+
+def _get_redis_dict(roach_num):
+    import redis
+
+    r = redis.Redis(redis_dbname)
+
+    # redis key names have the form "roachsensor:<roachhostname>"
+    # roachhostnames are pf1, pf2, ..., pf8
+    # Each key stores a hash table of different sensors
+    roach = "pf%d" % roach_num
+    rkey = "roachsensor:%s" % roach
+
+    # Get the entire hash table for this ROACH's key, returned as a dictionary
+    return r.hgetall(rkey)
+
+
+def create_from_redis(redis_dict=None):
+    """
+    Return a list of roach temperature objects from the redis database.
+
+    Parameters:
+    ------------
+    redis_dict: A dict spoofing the return dict from _get_redis_dict for testing
+        purposes. Default: None
+
+    Returns:
+    -----------
+    A list of RoachTemperature objects
+    """
+
+    roach_obj_list = []
+    for i in range(1, 9):
```

Comment:
This feels like something that shouldn't be hardcoded here.

Reply:
yeah, I agree, I just used some code from Jack. @jack-h can you suggest something better?

Reply:
It's not much better, but I moved the hardcoded list of roach names to the top of roach.py. I'm happy for other suggestions or we can just go with this.